apache-kvrocks fails to build from source with GCC 15 on amd64

The package apache-kvrocks with version 2.12.1-r1 fails to build from source with GCC 15 on amd64 with the following error:

```2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:124:1: error: no declaration matches 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, uint32_t, rocksdb::PinnableSlice&&, rocksdb::PinnableSlice&&, rocksdb::PinnableSlice&&, uint64_t)'
2025/05/31 00:00:37 INFO   124 | IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(
2025/05/31 00:00:37 INFO       | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:172:7: note: candidates are: 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(rocksdb::IteratorSeekQueryTraceRecord&)'
2025/05/31 00:00:37 INFO   172 | class IteratorSeekQueryTraceRecord : public IteratorQueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:190:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, const std::string&, const std::string&, const std::string&, int)'
2025/05/31 00:00:37 INFO   190 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:186:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, rocksdb::PinnableSlice&&, rocksdb::PinnableSlice&&, rocksdb::PinnableSlice&&, int)'
2025/05/31 00:00:37 INFO   186 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:183:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, const std::string&, int)'
2025/05/31 00:00:37 INFO   183 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:180:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, rocksdb::PinnableSlice&&, int)'
2025/05/31 00:00:37 INFO   180 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:172:7: note: 'class rocksdb::IteratorSeekQueryTraceRecord' defined here
2025/05/31 00:00:37 INFO   172 | class IteratorSeekQueryTraceRecord : public IteratorQueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:134:1: error: no declaration matches 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, uint32_t, const std::string&, const std::string&, const std::string&, uint64_t)'
2025/05/31 00:00:37 INFO   134 | IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(
2025/05/31 00:00:37 INFO       | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:172:7: note: candidates are: 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(rocksdb::IteratorSeekQueryTraceRecord&)'
2025/05/31 00:00:37 INFO   172 | class IteratorSeekQueryTraceRecord : public IteratorQueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:190:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, const std::string&, const std::string&, const std::string&, int)'
2025/05/31 00:00:37 INFO   190 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:186:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, rocksdb::PinnableSlice&&, rocksdb::PinnableSlice&&, rocksdb::PinnableSlice&&, int)'
2025/05/31 00:00:37 INFO   186 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:183:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, const std::string&, int)'
2025/05/31 00:00:37 INFO   183 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:180:3: note:                 'rocksdb::IteratorSeekQueryTraceRecord::IteratorSeekQueryTraceRecord(SeekType, int, rocksdb::PinnableSlice&&, int)'
2025/05/31 00:00:37 INFO   180 |   IteratorSeekQueryTraceRecord(SeekType seekType, uint32_t column_family_id,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:172:7: note: 'class rocksdb::IteratorSeekQueryTraceRecord' defined here
2025/05/31 00:00:37 INFO   172 | class IteratorSeekQueryTraceRecord : public IteratorQueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:155:10: error: no declaration matches 'uint32_t rocksdb::IteratorSeekQueryTraceRecord::GetColumnFamilyID() const'
2025/05/31 00:00:37 INFO   155 | uint32_t IteratorSeekQueryTraceRecord::GetColumnFamilyID() const {
2025/05/31 00:00:37 INFO       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:155:10: note: no functions named 'uint32_t rocksdb::IteratorSeekQueryTraceRecord::GetColumnFamilyID() const'
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:172:7: note: 'class rocksdb::IteratorSeekQueryTraceRecord' defined here
2025/05/31 00:00:37 INFO   172 | class IteratorSeekQueryTraceRecord : public IteratorQueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:168:1: error: no declaration matches 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(std::vector<unsigned int>, std::vector<rocksdb::PinnableSlice>&&, uint64_t)'
2025/05/31 00:00:37 INFO   168 | MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(
2025/05/31 00:00:37 INFO       | ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:220:7: note: candidates are: 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(const rocksdb::MultiGetQueryTraceRecord&)'
2025/05/31 00:00:37 INFO   220 | class MultiGetQueryTraceRecord : public QueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:226:3: note:                 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(int, const std::vector<std::__cxx11::basic_string<char> >&, int)'
2025/05/31 00:00:37 INFO   226 |   MultiGetQueryTraceRecord(std::vector<uint32_t> column_family_ids,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:222:3: note:                 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(int, std::vector<rocksdb::PinnableSlice>&&, int)'
2025/05/31 00:00:37 INFO   222 |   MultiGetQueryTraceRecord(std::vector<uint32_t> column_family_ids,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:220:7: note: 'class rocksdb::MultiGetQueryTraceRecord' defined here
2025/05/31 00:00:37 INFO   220 | class MultiGetQueryTraceRecord : public QueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:175:1: error: no declaration matches 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(std::vector<unsigned int>, const std::vector<std::__cxx11::basic_string<char> >&, uint64_t)'
2025/05/31 00:00:37 INFO   175 | MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(
2025/05/31 00:00:37 INFO       | ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:220:7: note: candidates are: 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(const rocksdb::MultiGetQueryTraceRecord&)'
2025/05/31 00:00:37 INFO   220 | class MultiGetQueryTraceRecord : public QueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:226:3: note:                 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(int, const std::vector<std::__cxx11::basic_string<char> >&, int)'
2025/05/31 00:00:37 INFO   226 |   MultiGetQueryTraceRecord(std::vector<uint32_t> column_family_ids,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:222:3: note:                 'rocksdb::MultiGetQueryTraceRecord::MultiGetQueryTraceRecord(int, std::vector<rocksdb::PinnableSlice>&&, int)'
2025/05/31 00:00:37 INFO   222 |   MultiGetQueryTraceRecord(std::vector<uint32_t> column_family_ids,
2025/05/31 00:00:37 INFO       |   ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:220:7: note: 'class rocksdb::MultiGetQueryTraceRecord' defined here
2025/05/31 00:00:37 INFO   220 | class MultiGetQueryTraceRecord : public QueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc: In destructor 'virtual rocksdb::MultiGetQueryTraceRecord::~MultiGetQueryTraceRecord()':
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:188:11: error: request for member 'clear' in '((rocksdb::MultiGetQueryTraceRecord*)this)->rocksdb::MultiGetQueryTraceRecord::cf_ids_', which is of non-class type 'int'
2025/05/31 00:00:37 INFO   188 |   cf_ids_.clear();
2025/05/31 00:00:37 INFO       |           ^~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc: At global scope:
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/trace_replay/trace_record.cc:192:23: error: no declaration matches 'std::vector<unsigned int> rocksdb::MultiGetQueryTraceRecord::GetColumnFamilyIDs() const'
2025/05/31 00:00:37 INFO   192 | std::vector<uint32_t> MultiGetQueryTraceRecord::GetColumnFamilyIDs() const {
2025/05/31 00:00:37 INFO       |                       ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:235:33: note: candidate is: 'virtual int rocksdb::MultiGetQueryTraceRecord::GetColumnFamilyIDs() const'
2025/05/31 00:00:37 INFO   235 |   virtual std::vector<uint32_t> GetColumnFamilyIDs() const;
2025/05/31 00:00:37 INFO       |                                 ^~~~~~~~~~~~~~~~~~
2025/05/31 00:00:37 INFO /home/build/output/_deps/rocksdb-src/include/rocksdb/trace_record.h:220:7: note: 'class rocksdb::MultiGetQueryTraceRecord' defined here
2025/05/31 00:00:37 INFO   220 | class MultiGetQueryTraceRecord : public QueryTraceRecord {
2025/05/31 00:00:37 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:01:19 WARN ninja: subcommands failed
2025/05/31 00:01:19 INFO
2025/05/31 00:01:19 ERRO failed to build package: unable to run package apache-kvrocks pipeline: unable to run pipeline: unable to run pipeline: exit status 1
make[1]: *** [Makefile:157: packages/x86_64/apache-kvrocks-2.12.1-r1.apk] Error 1
make[1]: Leaving directory '/home/sergio_durigan_chainguard_dev/os'
make: *** [Makefile:151: package/apache-kvrocks] Error 2
```

--- a/apache-kvrocks.yaml
+++ b/apache-kvrocks.yaml
@@ -1,7 +1,7 @@
 package:
   name: apache-kvrocks
   version: "2.13.0"
-  epoch: 0
+  epoch: 1
   description: Apache Kvrocks is a distributed key value NoSQL database that uses RocksDB as storage engine and is compatible with Redis protocol.
   copyright:
     - license: Apache-2.0
@@ -16,7 +16,6 @@ environment:
       - busybox
       - cmake
       - coreutils
-      - gcc-14-default
       - libtool
       - lld
       - openssl-dev

--- a/binutils.yaml
+++ b/binutils.yaml
@@ -2,7 +2,7 @@
 package:
   name: binutils
   version: "2.45"
-  epoch: 0
+  epoch: 1
   description: "GNU binutils"
   copyright:
     - license: GPL-2.0
@@ -18,7 +18,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - flex
-      - gcc-14-default
       - isl
       - posix-libc-utils
       - texinfo

--- a/bluez.yaml
+++ b/bluez.yaml
@@ -2,7 +2,7 @@
 package:
   name: bluez
   version: "5.83"
-  epoch: 2
+  epoch: 3
   description: Tools for the Bluetooth protocol stack
   copyright:
     - license: GPL-2.0 AND LGPL-2.1 AND BSD-2-Clause AND MIT
@@ -18,7 +18,6 @@ environment:
       - dbus
       - dbus-dev
       - ell-dev
-      - gcc-14-default
       - glib-dev
       - icu-dev
       - json-c-dev

--- a/cifs-utils.yaml
+++ b/cifs-utils.yaml
@@ -1,7 +1,7 @@
 package:
   name: cifs-utils
   version: "7.4"
-  epoch: 4
+  epoch: 5
   description: CIFS filesystem user-space tools
   copyright:
     - license: GPL-3.0-or-later
@@ -18,7 +18,6 @@ environment:
       - automake
       - build-base
       - busybox
-      - gcc-14-default
       - keyutils-dev
       - krb5-dev
       - libcap-dev

--- a/difftastic.yaml
+++ b/difftastic.yaml
@@ -1,7 +1,7 @@
 package:
   name: difftastic
   version: "0.64.0"
-  epoch: 1
+  epoch: 2
   description: "a structural diff that understands syntax"
   copyright:
     - license: MIT
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cargo-auditable
-      - gcc-14-default
       - rust
 
 pipeline:

--- a/envoy-gateway.yaml
+++ b/envoy-gateway.yaml
@@ -1,7 +1,7 @@
 package:
   name: envoy-gateway
   version: "1.5.0"
-  epoch: 1 # GHSA-f9f8-9pmf-xv68
+  epoch: 2
   description: Manages Envoy Proxy as a Standalone or Kubernetes-based Application Gateway
   copyright:
     - license: Apache-2.0
@@ -10,7 +10,6 @@ environment:
   contents:
     packages:
       - btrfs-progs-dev
-      - gcc-14-default
 
 pipeline:
   - uses: git-checkout

--- a/gitness.yaml
+++ b/gitness.yaml
@@ -1,7 +1,7 @@
 package:
   name: gitness
   version: "3.3.0"
-  epoch: 0 # CVE-2025-47907
+  epoch: 1
   description: Gitness is an Open Source developer platform with Source Control management, Continuous Integration and Continuous Delivery.
   copyright:
     - license: Apache-2.0
@@ -12,7 +12,6 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
-      - gcc-14-default
       - go~1.24
       - nodejs
       - protobuf

--- a/grafana-alloy.yaml
+++ b/grafana-alloy.yaml
@@ -1,7 +1,7 @@
 package:
   name: grafana-alloy
   version: "1.10.2"
-  epoch: 1 # GHSA-2464-8j7c-4cjm
+  epoch: 2
   description: OpenTelemetry Collector distribution with programmable pipelines
   copyright:
     - license: Apache-2.0
@@ -17,7 +17,6 @@ environment:
     packages:
       - busybox
       - ca-certificates-bundle
-      - gcc-14-default
       - go
       - nodejs-18
       - systemd-dev

--- a/imath.yaml
+++ b/imath.yaml
@@ -1,7 +1,7 @@
 package:
   name: imath
   version: "3.2.1"
-  epoch: 1
+  epoch: 2
   description: C++ and python library of 2D and 3D vector, matrix, and math operations for computer graphics
   copyright:
     - license: BSD-3-Clause
@@ -16,7 +16,6 @@ environment:
       - ca-certificates-bundle
       - cmake-3
       - doxygen
-      - gcc-14-default
       - py3-supported-numpy~1.26
       - python3-dev
       - samurai

--- a/jq.yaml
+++ b/jq.yaml
@@ -1,7 +1,7 @@
 package:
   name: jq
   version: "1.8.1"
-  epoch: 2
+  epoch: 3
   description: "a lightweight and flexible JSON processor"
   copyright:
     - license: MIT
@@ -13,7 +13,6 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
-      - gcc-14-default
       - libtool
       - oniguruma-dev
       - wolfi-base

--- a/krb5.yaml
+++ b/krb5.yaml
@@ -1,7 +1,7 @@
 package:
   name: krb5
   version: "1.22.1"
-  epoch: 0
+  epoch: 1
   description: The Kerberos network authentication system
   copyright:
     - license: MIT
@@ -24,7 +24,6 @@ environment:
       - ca-certificates-bundle
       - e2fsprogs-dev
       - flex
-      - gcc-14-default
       - keyutils-dev
       - krb5-conf
       - libverto-dev

--- a/kubeflow-pipelines.yaml
+++ b/kubeflow-pipelines.yaml
@@ -1,7 +1,7 @@
 package:
   name: kubeflow-pipelines
   version: "2.14.0"
-  epoch: 1 # GHSA-2464-8j7c-4cjm
+  epoch: 2
   description: Machine Learning Pipelines for Kubeflow
   checks:
     disabled:
@@ -17,7 +17,6 @@ environment:
     packages:
       - argo-cd
       - eslint
-      - gcc-14-default
       - go-licenses
       - jq
       - kubectl

